app/uk/gov/hmrc/gform/controllers/AuthenticatedRequestActions.scala

```diff
@@ -327,6 +365,19 @@ class AuthenticatedRequestActions(
         }
     }
 
+  override def authAndRetrieveForm[U <: SectionSelectorType: SectionSelector](
+    formTemplateId: FormTemplateId,
+    maybeAccessCode: Option[AccessCode],
+    operation: OperationWithForm
+  )(
+    f: Request[AnyContent] => LangADT => AuthCacheWithForm => SmartStringEvaluator => FormModelOptics[
+      DataOrigin.Mongo
+    ] => Future[Result]
+  ): Action[AnyContent] = withFormShutter(
+    formTemplateId,
+    authAndRetrieveFormNoShutter(formTemplateId, maybeAccessCode, operation)(f)
+  )
```

Comment:
You are right, piggybacking this on banner functionality is not right approach and 

> And with regards to json expectation, because we return Forbidden status it is not processed as success anyway.

also clears my concern about cases when we are retrieving json etc.

But I'm still not happy about us doing three round trips to backend for every request (and I'm not counting user form data fetching here).

1. Fetch the template json
2. Shutter data
3. Banner data

Can you explore feasibility of merging these three calls into single one? I think that may simplify code base a bit also.

Idea is to handle Shutter data and Banner data the same as we handle form template... ie. in controllers (and elsewhere) we are retrieving it from the request

```
val formTemplateWithRedirect = request.attrs(FormTemplateKey)
```

See SessionCookieDispatcherFilter.

Reply:
ok, will look into merging 3 calls into one...